sdk/tables/azure-tables/src/EdmGuid.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * A class that allows you to represent a GUID/UUID as a string.
+ */
+export class EdmGuid {
+  private _value: string = "";
+
+  constructor(guid: string) {
+    this.value = guid;
+  }
+
+  private validate(guid: string): boolean {
+    const re = /^(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}|00000000-0000-0000-0000-000000000000)$/i;
```

Comment:
we have a regex for this in `core-http` already: https://github.com/Azure/azure-sdk-for-js/blob/6661ad05d52ac3d050007bc2697557dfbe17b9eb/sdk/core/core-http/src/util/utils.ts#L85